src/lib/reflected-attribute-name.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * Elements and their special properties that should be reflected to attributes
+ * when set.
+ * 
+ * Modify this object to add reflected attributes.
+ *
+ * Each item in the array takes the following format:
+ *
+ * first: Element name or array of element names.
+ * rest:  Array of ([property name, reflected attribute name] or
+ *        property name (if reflected attribute name is identical).
+ */
+const reflectedAttributesSource: Array<{
+  tagNames: string[];
+  reflections: (string | string[])[];
+}> = [
+  {
+    tagNames: ['input', 'option'],
+    reflections: ['value']
+  },
+  {
+    tagNames: ['*'],
+    reflections: [['className', 'class'], 'id']
+  }
+];
+
+// reflectedAttributesSource is easy to visually parse, but we can reconfigure the
+// data structure into a map of maps for faster lookup at runtime.
+const reflectedAttributes = new Map<string, Map<string, string>>();
+const addReflectionForElement = (elementName: string, propertyName: string, attributeName: string) => {
+  if (reflectedAttributes.has(elementName)) {
+    reflectedAttributes.get(elementName)!.set(propertyName, attributeName);
+  } else {
+    reflectedAttributes.set(elementName, new Map([[propertyName, attributeName]]));
+  }
+};
+const addReflectionsForElement = (elementName: string, reflections: Array<string|Array<string>>) => {
+  for (const reflection of reflections) {
+    if (reflection instanceof Array) {
+      // Property has a different reflected attribute name.
+      addReflectionForElement(elementName, reflection[0], reflection[1]);
```

Comment:
Another personal style nit: I would probably have created local variables `propertyName` and `attributeName` and conditionally set them but have one call to `addReflectionForElement`. I think that's a bit more self-documenting. No real need to change though.